android/engine/src/main/java/org/smartregister/fhircore/engine/sync/SyncListenerManager.kt

```diff
@@ -53,6 +53,12 @@ constructor(
 
   private val syncConfig by lazy { configurationRegistry.getSyncConfigs() }
 
+  private fun isInitialSync() =
+    sharedPreferencesHelper
+      .read(SharedPreferenceKey.LAST_SYNC_TIMESTAMP.name, null)
+      .isNullOrBlank()
+      .not()
```

Comment:
Is initial sync is when the timestamp is null

https://github.com/d-tree-org/fhircore/blob/b5b91af98ed6fbc818d76b6ec71632ed2e35b13b/android/engine/src/main/java/org/smartregister/fhircore/engine/sync/SyncBroadcaster.kt#L112-L113

Reply:
### I don't understand what this means
> Is initial sync is when the timestamp is null

Reply:
The timestamp is set after the initial sync, so if you want to check if a sync is an initial sync you have to check if the timestamp is null. On that line, you are negating which means this is more of a check if the sync is subsequent than the initial one. Check the implementation in `SyncBroadcaster.kt`